pkg/aws/plugin.go

```diff
@@ -0,0 +1,258 @@
+/*
+Copyright 2023 The Paraglider Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package aws
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"time"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/service/ec2"
+	"github.com/aws/aws-sdk-go-v2/service/ec2/types"
+	"github.com/paraglider-project/paraglider/pkg/paragliderpb"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+)
+
+type AwsPluginServer struct {
+	paragliderpb.UnimplementedCloudPluginServer
+	orchestratorServerAddr string
+}
+
+func (s *AwsPluginServer) CreateResource(ctx context.Context, req *paragliderpb.CreateResourceRequest) (*paragliderpb.CreateResourceResponse, error) {
+	return s._CreateResource(ctx, req, &awsClients{})
+}
+
+func (s *AwsPluginServer) _CreateResource(ctx context.Context, req *paragliderpb.CreateResourceRequest, awsClients *awsClients) (*paragliderpb.CreateResourceResponse, error) {
+	// Unmarshal resource description
+	runInstancesInput := &ec2.RunInstancesInput{}
+	err := json.Unmarshal(req.Description, runInstancesInput)
+	if err != nil {
+		return nil, fmt.Errorf("unable to marshal resource description: %w", err)
+	}
+
+	// Ensure resource description does not contain networking information
+	if len(runInstancesInput.NetworkInterfaces) > 0 {
+		return nil, fmt.Errorf("resource description should not contain network interfaces")
+	}
+	if len(runInstancesInput.SecurityGroupIds) > 0 || len(runInstancesInput.SecurityGroups) > 0 {
+		return nil, fmt.Errorf("resource description should not contain security groups")
+	}
+	if runInstancesInput.SubnetId != nil {
+		return nil, fmt.Errorf("resource description should not contain subnet ID")
+	}
+	if runInstancesInput.PrivateIpAddress != nil || runInstancesInput.EnablePrimaryIpv6 != nil ||
+		runInstancesInput.Ipv6AddressCount != nil || len(runInstancesInput.Ipv6Addresses) > 0 {
+		return nil, fmt.Errorf("resource description should not contain IP address information")
+	}
+
+	// Get region
+	availabilityZone := *runInstancesInput.Placement.AvailabilityZone
+	region := availabilityZone[:len(availabilityZone)-1]
```

Comment:
Maybe this should wait until there's more implemented, but might this be helpful to have in a helper function?